pipeline/src/node_properties/snapshot_properties.rs

```diff
@@ -3,13 +3,35 @@ use serde_with::skip_serializing_none;
 use std::path::PathBuf;
 use url::Url;
 
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(tag = "location", rename_all = "snake_case")]
+pub enum SnapshotProperties {
+    Local(LocalSnapshotProperties),
+    LumeoCloud(LumeoCloudSnapshotProperties),
+}
+
 #[skip_serializing_none]
-#[derive(Debug, Default, Clone, PartialEq, Serialize, Deserialize)]
-pub struct SnapshotProperties {
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct CommonSnapshotProperties {
+    pub min_duration: Option<u64>,
+    pub max_duration: Option<u64>,
+    pub max_size: Option<u64>,
+    pub retention_duration: Option<u64>,
+    pub webhook_url: Option<Url>,
+    pub trigger: Option<String>,
+}
+
+#[skip_serializing_none]
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct LocalSnapshotProperties {
+    #[serde(flatten)]
+    pub common: CommonSnapshotProperties,
     pub path: Option<PathBuf>,
     pub max_edge_files: Option<u64>,
```

Comment:
perhaps we should support max_edge_files in either case so that we don't fill up local space in case uploads are slow / there's a network disconnection ?

Reply:
That would be a separate change; if we do it we should do it for both clips & snapshots. @zeenix wdyt?

Reply:
yeah can be separate change if we don't have this for clips either.